src/encryption.rs

```diff
@@ -8,18 +8,46 @@
 
 // TODO(dirvine) Look at aessafe 256X8 cbc it should be very much faster  :01/03/2015
 
-use rust_sodium::crypto::secretbox::{self, KEYBYTES, NONCEBYTES};
+use crate::sequential::{Iv, Key};
+use crate::MAX_CHUNK_SIZE;
+use crate::{SelfEncryptionError, StorageError};
+use aes::Aes128;
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
+type Aes128Cbc = Cbc<Aes128, Pkcs7>;
 
-pub use rust_sodium::crypto::secretbox::{Key, Nonce as Iv};
-pub type DecryptionError = ();
+pub const KEY_SIZE: usize = 16;
+pub const IV_SIZE: usize = 16;
 
-pub const KEY_SIZE: usize = KEYBYTES;
-pub const IV_SIZE: usize = NONCEBYTES;
+// Buffer size is set to max chunk size + 100 bytes for padding.
```

Comment:
Do we need to consider 100 bytes here for padding. It should be no more than 15 bytes I htink, but taken care of by the block_modes crate AFAIK.

Reply:
Or use [`encrypt_vec`](https://docs.rs/block-modes/0.3.3/block_modes/trait.BlockMode.html#method.encrypt_vec) (and `decrypt_vec`) and you don't have to worry about the padding at all.